### PR TITLE
feat: data table mvp

### PR DESCRIPTION
**Pull request by danielrobertson**
Closes # 

## 📝 Description

- `DataTable` is a table including 
  - sort callback
  - pagination and page change callback 
  - sticky left and right columns, 1 per side
- Prerelease version 

### Storybook
https://user-images.githubusercontent.com/3459902/228329658-3984e9e9-2edf-485d-a556-6e407a673b6b.mov

### Docs site 
image

## Merge checklist

- [x] Added/updated tests
- [x] Added changeset

### CODE REVIEW

File: packages/react/src/components/DataTable/DataTable.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        console.warn('Only one column can be pinned right');
+      }
+    }
+  }, [columns]);
+
+  return (
+    <StyledDataTable {...other} ref={forwardedRef} as={as} className={className} css={css}>
+      <DataTableContext.Provider value={context}>
+        <div style={{ overflowX: 'auto' }}>
+          <table className={table()} style={{ borderCollapse: 'separate' }}>
```
Comment (danielrobertson) [inline packages/react/src/components/DataTable/DataTable.tsx:66]: Looking for suggestions most appropriate way to Stitches-ify this `table` and its parent `div`

Comment (bclark-p44) [reply packages/react/src/components/DataTable/DataTable.tsx:66]: These can just be `styled` components as well.

Comment (danielrobertson) [reply packages/react/src/components/DataTable/DataTable.tsx:66]: This means I have to _name them_! The horror

File: apps/website/docs/data-display/data-table.mdx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+## Usage
+
+### Pinned columns
+
+The first and last columns can be pinned with `isPinnedLeft` and `isPinnedRight` fields in the
+column.
+
+```jsx live
+function DataTableExample() {
+  const columnDefinitions = [
```
Comment (danielrobertson) [inline apps/website/docs/data-display/data-table.mdx:112]: Was browsing docusaurus hoping to find a way to extract this shared data fixture out, but not seeing anything. https://docusaurus.io/docs/markdown-features/code-blocks

Comment (danielrobertson) [reply apps/website/docs/data-display/data-table.mdx:112]: alternative, add faker-js dependency to just generate random data and cut down on the visual overload of all this JSON just to demonstrate a meaningful data table

File: packages/css/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "../../../.moon/cache/types/packages/system/css"
+    "outDir": "../../.moon/cache/types/packages/css"
```
Comment (danielrobertson) [inline packages/css/tsconfig.json:4]: something automated changed this

File: packages/react/src/components/DataTable/DataTable.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    onClick={column.isSortable ? handleSort(column.key) : undefined}
+                  >
+                    {column.header}
+                  </DataTableColumn>
+                ))}
+              </DataTableRow>
+            </DataTableHeader>
+            <DataTableBody>
+              {rows.map((row, idx) => (
+                <DataTableRow key={`row-${String(idx)}`}>
```
Comment (danielrobertson) [inline packages/react/src/components/DataTable/DataTable.tsx:88]: open to suggestions on how not to do this. uuid or req some row schema that guarantees a key/id

**Comment by danielrobertson** [comment]
_Note on prerelease version_:

Locally, I ran `yarn changeset pre enter data-table` to tag a prerelease `data-table.0` version. This created `pre.json` which `yarn version` will use to associate the changeset (`.changeset/forty-apricots-deny.md` in this PR) and bump the versions of storybook, website, and react as shown in the attached pictures 
image
image
image

In the future when ready to take data-table out of prerelease and into release, we'll follow the `yarn changeset pre exit` instructions here and commit another changeset https://github.com/changesets/changesets/blob/main/docs/prereleases.md

**Pull request closed without merging**